projects/eGolf/Canvas.java

```diff
@@ -1,59 +1,154 @@
-  private int radius = 5;
-  private Point center = new Point(getSize().width/2, getSize().height/2);
+  private HashMap<Point, Integer> wheels = new HashMap<>();
+  private int radius = 15;
+  private Point2D.Double center;
+  private Point2D.Double vect;
+  private boolean draggingVect = false;
+  private boolean draggingWheel = false;
+  private Point wheelStart;
+  private int frameRate = 60;
+  private double friction = 0.99; // 1 = no loss, 0 = loses all in one frame
+
   public Canvas() {
-    // 17 milliseconds is roughly 60fps
-    //Timer timer = new Timer(17, (ActionEvent e) -> { repaint(); });
-    //timer.start();
+    center = new Point2D.Double(getSize().width / 2, getSize().height / 2); // this is broken
+    vect = new Point2D.Double(Math.random() * 2000, Math.random() * 2000);
+    Timer timer = new Timer(1000 / frameRate, (ActionEvent e) -> {
+      perFrame();
+    });
+    timer.start();
   }
-
+  
   @Override
-    protected void paintComponent(Graphics g) {
-      super.paintComponent(g);
-      var d = getSize();
-      
-      g.setColor(Color.DARK_GRAY);
-      g.fillRect(0, 0, d.width, d.height);
-      g.setColor(Color.RED);
-      g.drawOval(center.x-radius/2, center.y-radius/2-24, radius, radius);
-  }
+  protected void paintComponent(Graphics g) {
+    super.paintComponent(g);
+    var d = getSize();
+    Point mouseLoc = MouseInfo.getPointerInfo().getLocation();
 
-    @Override
-    public void mouseClicked(MouseEvent e) {
-      center.x = e.getX();
-      center.y = e.getY();
-      repaint();
+    // background
+    g.setColor(Color.DARK_GRAY);
+    g.fillRect(0, 0, d.width, d.height);
+    // center
+    g.setColor(Color.white);
+    g.drawOval((int) center.x - 3, (int) center.y - 3, 6, 6);
+    // outer edge
+    g.setColor(Color.RED);
+    g.drawOval((int) center.x - radius, (int) center.y - radius, radius * 2, radius * 2);
+    // drag vect
+    g.setColor(Color.green);
+    if (draggingVect) {
+      g.drawLine((int) center.x, (int) center.y,
+          (int) (center.x + 2 * (center.x - mouseLoc.x)), // stole from stack overflow
+          (int) (center.y + 2 * (center.y - mouseLoc.y)) - 30);
     }
 
-    @Override
-    public void mousePressed(MouseEvent e) {
-      System.out.println("pressed");
+    // motion in 1s
+    g.setColor(Color.blue);
+    g.drawLine((int) center.x, (int) center.y, (int) (center.x + vect.x), (int) (center.y + vect.y));
+    // new wheel
+
+    if (draggingWheel) {
+      int r = twoDist(mouseLoc, wheelStart);
+      Point tl = topLeft(wheelStart, radius);
+      g.setColor(Color.yellow);
+      g.drawOval(tl.x, tl.y, r, r);
     }
+    // drawing wheels
 
-    @Override
-    public void mouseReleased(MouseEvent e) {
-      System.out.println("releaased");
+    for (Map.Entry<Point, Integer> entry : wheels.entrySet()) {
+      Point center = entry.getKey();
+      int rad = entry.getValue();
+      g.setColor(Color.orange);
```

Comment:
same thing here, if you are just declaring variables once maybe just put everything on 1 line to save space.